api/security/api-key-authorize.js

```diff
@@ -0,0 +1,88 @@
+'use strict'
+
+const {ApiKey} = require('../../models/db')
+const {forEach} = require('lodash')
+
+module.exports = function authorize (request, response, next) {
+  validate(request, function (error, availableScopes) {
+    if (!error) {
+      if (!availableScopes || !availableScopes.length) {
+        next(buildUnauthorized(error))
+      } else {
+        var requestedResource
+
+        if (request.method === 'GET') {
+          requestedResource = request.query['uri']
+        } else {
+          requestedResource = request.body['uri']
+        }
+        if (isAuthorized(availableScopes, requestedResource)) {
+          next()
+        } else {
+          next(buildUnauthorized(error))
+        }
+      }
+    } else {
+      next(error)
+    }
+  })
+}
+
+function isAuthorized (availableScopes, requestedResource) {
+  var authorized = false
+
+  if (Array.isArray(availableScopes)) {
+    forEach(availableScopes, (scope) => {
```

Comment:
Instead of forEach, you could use [find](https://lodash.com/docs/4.17.5#filter) which iterates over the collection, returning the first that matches the predicate, or undefined if none match. Basically short circuits when it finds the first.